tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go

```diff
@@ -0,0 +1,382 @@
+                key: postgresql_conn_str
+      restartPolicy: Never
+  backoffLimit: 4
+`
+
+	insertRecordsJobTemplate = `apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    app: postgresql-insert-job
+  name: postgresql-insert-job
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    metadata:
+      labels:
+        app: postgresql-insert-job
+    spec:
+      containers:
+      - image: ghcr.io/kedacore/tests-postgresql
+        imagePullPolicy: Always
+        name: postgresql-processor-test
+        command:
+          - /app
+          - insert
+        env:
+          - name: TASK_INSTANCES_COUNT
+            value: "10000"
+          - name: CONNECTION_STRING
+            valueFrom:
+              secretKeyRef:
+                name: {{.SecretName}}
+                key: postgresql_conn_str
+      restartPolicy: Never
+  backoffLimit: 4
+`
+)
+
+func TestPostreSQLScaler(t *testing.T) {
+	// Create kubernetes resources for PostgreSQL server
+	kc := GetKubernetesClient(t)
+	data, postgreSQLtemplates := getPostgreSQLTemplateData()
+
+	CreateKubernetesResources(t, kc, testNamespace, data, postgreSQLtemplates)
+	hashiCorpToken := setupHashiCorpVault(t, kc)
+
+	assert.True(t, WaitForStatefulsetReplicaReadyCount(t, kc, postgreSQLStatefulSetName, testNamespace, 1, 60, 3),
+		"replica count should be %d after 3 minutes", 1)
+
+	createTableSQL := "CREATE TABLE task_instance (id serial PRIMARY KEY,state VARCHAR(10));"
+	ok, out, errOut, err := WaitForSuccessfulExecCommandOnSpecificPod(t, postgresqlPodName, testNamespace,
+		fmt.Sprintf("psql -U %s -d %s -c \"%s\"", postgreSQLUsername, postgreSQLDatabase, createTableSQL), 60, 3)
+	assert.True(t, ok, "executing a command on PostreSQL Pod should work; Output: %s, ErrorOutput: %s, Error: %s", out, errOut, err)
+
+	// Create kubernetes resources for testing
+	data, templates := getTemplateData()
+	data.HashiCorpToken = hashiCorpToken
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	testActivation(t, kc, data)
+	testScaleUp(t, kc, data)
+	testScaleDown(t, kc)
+
+	// cleanup
+	KubectlDeleteMultipleWithTemplate(t, data, templates)
+	cleanupHashiCorpVault(t, kc)
+	DeleteKubernetesResources(t, kc, testNamespace, data, postgreSQLtemplates)
+}
+
+func setupHashiCorpVault(t *testing.T, kc *kubernetes.Clientset) string {
+	CreateNamespace(t, kc, vaultNamespace)
+
+	_, err := ExecuteCommand("helm repo add hashicorp https://helm.releases.hashicorp.com")
+	assert.NoErrorf(t, err, "cannot add hashicorp repo - %s", err)
+
+	_, err = ExecuteCommand("helm repo update")
+	assert.NoErrorf(t, err, "cannot update repos - %s", err)
+
+	_, err = ExecuteCommand(fmt.Sprintf(`helm upgrade --install --set "server.dev.enabled=true" --namespace %s --wait vault hashicorp/vault`, vaultNamespace))
```

Comment:
remove the quotes around `"server.dev.enabled=true"`. I think that using our approach, it makes something weird. At least I had this problem with datadog chart

Reply:
In fact, if you check the chart values when you install it with quotes, you will see that the value is not parsed and it's registered as another different value with the quotes. At least, that was my case, all the values with quotes weren't parsed and was registered wrong

Reply:
I thought so the problem was with the way it was being parsed, but I couldn't figure out how. I ran a [snippet](https://go.dev/play/p/7lFsBrrjOpZ) on the go playground, and it splits the way I expect it to. Not sure what the issue is. Maybe something in the internals of the `exec` package and the way it executes the command?

Reply:
Maybe... Have you tried without the quotes? Does it work?

Reply:
Yes, it does as I mentioned above. I am just not sure why :P

Reply:
This is [the line](https://github.com/kedacore/keda/blob/main/tests/scalers_go/datadog/datadog_test.go#L303) I talked about

Reply:
I think that it doesn't like the command and all agrs together in one `string`, it should be probably `[]string` passed to the `exec`. I do recall having some problems around this somewhere in the past 😅

Reply:
@zroubalik This is what is being done in the helper package. The `string` is split into `string[0]` and `string[1:]` and passed to `exec`.

Reply:
Yeah, the exec accepts `[]string`, so there's a high chance that it is causing the problems:
https://github.com/kedacore/keda/blob/17b78dd66f63ec58981facd1b78aad5114302456/tests/helper/helper.go#L113-L124

Reply:
https://github.com/kedacore/keda/blob/17b78dd66f63ec58981facd1b78aad5114302456/tests/helper/helper.go#L64-L79

I'm referring to this.

Reply:
Ah right, I thought that this command is run on a pod.

Reply:
https://go.dev/play/p/bBitQgGZ7qk

Running this snippet and printing the `exec.Cmd` object prints the expected command. I am so confused right now.